core/src/types/blocking_read/blocking_reader.rs

```diff
@@ -18,18 +18,20 @@
 use std::collections::Bound;
 use std::ops::Range;
 use std::ops::RangeBounds;
+use std::sync::Arc;
 
-use bytes::Buf;
 use bytes::BufMut;
 
-use crate::raw::oio::BlockingRead;
 use crate::raw::*;
 use crate::*;
 
+use super::buffer_iterator::BufferIterator;
+
 /// BlockingReader is designed to read data from given path in an blocking
 /// manner.
 pub struct BlockingReader {
-    pub(crate) inner: oio::BlockingReader,
+    pub(crate) inner: Arc<dyn oio::BlockingRead>,
```

Comment:
Please change `oio::BlockingReader` to `Arc<dyn oio::BlockingRead>`, maybe worth a seperate PR.

Reply:
Would you like to submit this PR first?